galleries/examples/ticks/date_formatters_locators.py

```diff
@@ -12,12 +12,7 @@
 import matplotlib.pyplot as plt
 import numpy as np
 
-from matplotlib.dates import (FR, MO, MONTHLY, SA, SU, TH, TU, WE,
-                              AutoDateFormatter, AutoDateLocator,
-                              ConciseDateFormatter, DateFormatter, DayLocator,
-                              HourLocator, MicrosecondLocator, MinuteLocator,
-                              MonthLocator, RRuleLocator, SecondLocator,
-                              WeekdayLocator, YearLocator, rrulewrapper)
+from matplotlib.dates import DateFormatter
```

Comment:
It seems that while these aren't referenced in the code directly, they are referenced from strings that get evaled:

```
        Traceback (most recent call last):
          File "/home/circleci/project/galleries/examples/ticks/date_formatters_locators.py", line 58, in <module>
            plot_axis(ax, locator, xmax, fmt)
          File "/home/circleci/project/galleries/examples/ticks/date_formatters_locators.py", line 27, in plot_axis
            ax.xaxis.set_major_locator(eval(locator))
                                       ^^^^^^^^^^^^^
          File "<string>", line 1, in <module>
        NameError: name 'AutoDateLocator' is not defined
```

Reply:
Sigh, `isort` wants to remove the `noqa`, while `ruff`/`flake8` wants it. I'll just restore the exception then.